test/unit/spec/Registration_spec.js

```diff
@@ -994,5 +996,45 @@ Expect.describe('Registration', function() {
         expectRegApiError(test, '\'Email address \' must be in the form of an email address');
       });
     });
+    itp('tests primary button staying disabled through save state', function() {
+      const parseSchemaSpy = jasmine.createSpy('parseSchemaSpy');
+      const preSubmitSpy = jasmine.createSpy('preSubmitSpy');
+      const postSubmitSpy = jasmine.createSpy('postSubmitSpy');
+      const setting = {
+        registration: {
+          parseSchema: function(resp, onSuccess, onFailure) {
+            parseSchemaSpy(resp, onSuccess, onFailure);
+            onSuccess(resp);
+          },
+          preSubmit: function(postData, onSuccess, onFailure) {
+            preSubmitSpy(postData, onSuccess, onFailure);
+            onSuccess(postData);
+          },
+          postSubmit: function(postData, onSuccess, onFailure) {
+            postSubmitSpy(postData, onSuccess, onFailure);
+            onSuccess(postData);
+          },
+        },
+      };
+
+      return setup(setting)
+        .then(function(test) {
+          spyOn(Backbone.Model.prototype, 'save').and.returnValue($.Deferred().resolve());
+          spyOn(Settings.prototype, 'postRegistrationSubmit').and
+            .returnValue(new Promise(resolve => setTimeout(resolve, 105)));
+          Util.resetAjaxRequests();
+          test.form.setUserName('test@example.com');
+          test.form.setPassword('Abcd1234');
+          test.form.setFirstname('firstName');
+          test.form.setLastname('lastName');
+          test.form.setReferrer('referrer');
+          expect($('input.button-primary').length).toBe(1);
+          expect($('input.button-primary.btn-disabled').length).toBe(0);
+          test.form.submit();
+          return new Promise(resolve => setTimeout(resolve, 100));
+        }).then(() => {
+          expect($('input.button-primary.btn-disabled').length).toBe(1);
+        });
+    });
```

Comment:
@conorhanrahan-okta The test has been added. Kindly review, thanks.

Reply:
Looks good to me!